lib/test.py

```diff
@@ -277,6 +277,8 @@ def run(self, server):
                                            "Test failed! Output from log file "
                                            "{0}:\n".format(log_file))
                 where = ": there were warnings in the valgrind log file(s)"
+            vardir = self.suite_ini['vardir']
+            os.rename(vardir, vardir + '.reject')
```

Comment:
1. I'm against of renaming of a working directory of a running instance. The result would be messy (for a `core = tarantool` test suite): xlog and text log files will continue be written into the old directory, new xlogs maybe will be written into the new one (not sure), restart would possibly lead to an error due to lack of the var directory. Please, don't do that.
2. I see two possible ways to avoid the rename and both ways are about recursive copying of worker's var directory, but at different moments:
    - When all tasks are finished (before exit from test-run), copy working directories of workers, which have failed tests, into an artefacts directory.
        - Upside: tarantool server is stopped, so we can be sure that files (say, log) are in some non-transient state. However it unlikely to occur and I guess is not critical.
        - Downside: next tests will populate xlogs, logs and other directory content when `--force` is enabled (it is so in our CI).
    - When a test failed, copy worker's working directory into an artefacts directory.
        - Upside: last xlog / log entries will be from the failed test.
        - Downside: All previous tests on the worker will left trails anyway. The only way to overcome it is to run each test inside its own directory (after #215 if we'll decide to implement it), but it is surely out of scope of this issue.
        - NB: Think re naming: what if two tests are failed on one worker? What if it is the same test, but on different configurations? Or even the same test with the same configuration: it is possible when we run test-run manually like `./test-run.py foo foo foo`. In the latter case we're not obligated to save all failed results (however we can, if it is easy), but test-run itself should not fail due to the artefacts directory existence. BTW, we can just check whether a directory exist and add .2, .3, etc suffix if so.
3. I propose the following directory name: `test/var/artefacts`.
4. There is nothing about storing job artefacts in https://github.com/tarantool/tarantool/issues/5050. Sasha, please, stop mixing different tasks into one. Those two (artefacts and reruns) are not even related to each other. There is #90 exactly about artefacts.

Reply:
Right, I've decided to use the second way, due to save artifacts should be the same as tests manually run results. In the first way results will be cut just after the fail and it can produce the questions from the users on their reruns, when they would get full results it it will be hardly compared. So the second way is the best I think.
I've added new ArtifactsWatcher class into the dispatcher listener which collects artifacts files of the failed tests on the testing finish. It copies the files from the failed tests to the '<vardir>/artifacts' directory to be able to collect these artifacts in gitlab-ci later.

Reply:
IMHO, in the best homes in London and Paris, the conversation resolves by the same person who started it.

Reply:
My usual agreement regarding who resolves PR comments is the following: feel free to resolve my comment youself if you're quite sure that the question is actually resolved. I'll reopen if something requires more attention. This way we'll as much as possible near to an actual state (what is done and what is not) at every moment.